src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js

```diff
@@ -0,0 +1,309 @@
+  },
+
+
+  initialize: function (options) {
+    L.CartoDBd3Layer.prototype.initialize.apply(this, [options]);
+    options = options || {};
+    // Set options
+    L.Util.setOptions(this, options);
+
+    this.fire = this.trigger;
+
+    CartoDBLayerCommon.call(this);
+    L.TileLayer.prototype.initialize.call(this);
+    this.interaction = [];
+    this.addProfiling();
+  },
+
+  addProfiling: function() {
+    this.bind('tileloadstart', function(e) {
+      var s = this.tileStats || (this.tileStats = {});
+      s[e.tile.src] = Profiler.metric('cartodb-js.tile.png.load.time').start();
+    });
+    var finish = function(e) {
+      var s = this.tileStats && this.tileStats[e.tile.src];
+      s && s.end();
+    };
+    this.bind('tileload', finish);
+    this.bind('tileerror', function(e) {
+      Profiler.metric('cartodb-js.tile.png.error').inc();
+      finish(e);
+    });
+  },
+
+
+  // overwrite getTileUrl in order to
+  // support different tiles subdomains in tilejson way
+  getTileUrl: function (tilePoint) {
+    var EMPTY_GIF = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7";
+    this._adjustTilePoint(tilePoint);
+
+    var tiles = [EMPTY_GIF];
+    if(this.tilejson) {
+      tiles = this.tilejson.tiles;
+    }
+
+    var index = (tilePoint.x + tilePoint.y) % tiles.length;
+
+    return L.Util.template(tiles[index], L.Util.extend({
+      z: this._getZoomForUrl(),
+      x: tilePoint.x,
+      y: tilePoint.y
+    }, this.options));
+  },
+
+  /**
+   * Change opacity of the layer
+   * @params {Integer} New opacity
+   */
+  setOpacity: function(opacity) {
+
+    if (isNaN(opacity) || opacity>1 || opacity<0) {
+      throw new Error(opacity + ' is not a valid value');
+    }
+
+    // Leaflet only accepts 0-0.99... Weird!
+    this.options.opacity = Math.min(opacity, 0.99);
+
+    if (this.options.visible) {
+      L.TileLayer.prototype.setOpacity.call(this, this.options.opacity);
+      this.fire('updated');
+    }
+  },
+
+
+  /**
+   * When Leaflet adds the layer... go!
+   * @params {map}
+   */
+  onAdd: function(map) {
+    // this.options.user = this.options.user_name;
+    // this.options.table = this.options.layer_definition.layers[0].options.layer_name;
+    var self = this;
+    this.options.map = map;
+
+    // Add cartodb logo
+    if (this.options.cartodb_logo != false)
+      CartoDBLogo.addWadus({ left:8, bottom:8 }, 0, map._container);
+    // TODO: We can probably move this to an initialize method
+    this.model.bind('change:urls', function() {
+      self.__update(function() {
+        // if while the layer was processed in the server is removed
+        // it should not be added to the map
+        var id = L.stamp(self);
+        if (!map._layers[id]) {
+          return;
+        }
+        L.CartoDBd3Layer.prototype.onAdd.apply(self, [map]);
+        L.CartoDBd3Layer.prototype.setCartoCSS.apply(self, [self.model.collection.models[1].layers.models[0].get("cartocss")]); 
```

Comment:
I'm totally lost here with `models[0]` and `models[1]` direct access. Can you elaborate a bit why it's like that :pray: